### PR TITLE
Upgrade mutable-argument-defaults to unsafe

### LINKED ISSUE #8104: B006 (mutable-argument-default) no longer offers autofix even with --unsafe-fixes
Hello 👋 

B006 (`mutable-argument-default`) no longer offers an autofix even when `--unsafe-fixes` is passed.

Use the example from `ruff rule B006` to reproduce this:

```python
# /tmp/bad.py
def add_to_list(item, some_list=[]):
    some_list.append(item)
    return some_list
```

At HEAD (df807ff91248c896cf7f11b25e4f64ecb1d77066) and v0.1.1:

```sh
❯ cargo run -p ruff_cli -- check --select B --unsafe-fixes /tmp/bad.py --no-cache
/tmp/bad.py:1:33: B006 Do not use mutable data structures for argument defaults
Found 1 error.
```

I've bisected this to 22e18741bdf58bc15a69da2503d7b48443e5a038, but that may only be where the issue was surfaced in the CLI.

The output is slightly different in that commit, there's an additional line:

```sh
/tmp/bad.py:1:33: B006 Do not use mutable data structures for argument defaults
Found 1 error.
[*] 0 fixable with the --fix option.
```

If I omit `--unsafe-fixes` there's a hint about a hidden fix:

```sh
❯ cargo run -p ruff_cli -- check --select B /tmp/bad2.py --no-cache
/tmp/bad.py:1:33: B006 Do not use mutable data structures for argument defaults
Found 1 error.
1 hidden fix can be enabled with the `--unsafe-fixes` option.
```

(I don't see this hint in 0.1.1 or HEAD, but I haven't bisected where it went away.)

In the parent of the bisected commit and in v0.0.292, I get the expected output:

```sh
/tmp/bad.py:1:33: B006 [*] Do not use mutable data structures for argument defaults
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

(I noticed after the fact that I ran Ruff from its repository which has Ruff config in pyproject.toml, so that could have skewed results a little. But the repro should still be valid.)

### PR DESCRIPTION
**Pull request by cjolowicz**
## Summary

Fixes #8104

## Test Plan

Covered by snapshot tests.

### CODE REVIEW

File: crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs [hunk 1]
```diff
@@ -197,5 +197,5 @@ fn move_initialization(
     }
 
     let initialization_edit = Edit::insertion(content, pos);
-    Some(Fix::display_edits(default_edit, [initialization_edit]))
+    Some(Fix::unsafe_edits(default_edit, [initialization_edit]))
```
Comment (zanieb) [inline crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs:200]: cc @charliermarsh 

Can we add a note here explaining _why_ the fix is unsafe? I'm not sure why it was marked as unapplicable originally as the fix looks relatively safe to me?

Comment (cjolowicz) [reply crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs:200]: Commented on this in the issue: https://github.com/astral-sh/ruff/issues/8104#issuecomment-1773834910

Comment (cjolowicz) [reply crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs:206]: I've added a `Known problems` section to explain why the fix is unsafe.

Comment (charliermarsh) [reply crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs:206]: Thank you!

**Pull request merged**